### PR TITLE
Silence C++ [[nodiscard]] warning

### PR DESCRIPTION
**Pull request by sonals**
Add a note about new aie-pdi-transform component

#### Problem solved by the commit
Tune clang-tidy behavior to not generate warnings about [[nodiscard]]

#### Bug / issue (if any) fixed, which PR introduced the bug, how it was discovered

1. Silence the [[nodiscard]] clang-tidy messages
2. Add a note about aie-pdi-transform

#### How problem was solved, alternative solutions (if any) and why they were rejected
Updated .clang-tidy rules

#### Risks (if any) associated the changes in the commit
None

#### What has been tested and how, request additional testing if necessary
Compile tested

#### Documentation impact (if any)
None

### CODE REVIEW

File: src/runtime_src/hip/core/memory.cpp [hunk 1]
```diff
@@ -103,7 +103,7 @@ namespace xrt::core::hip
         assert(src_hip_mem->get_flags() == hipHostMallocDefault || src_hip_mem->get_flags() == hipHostMallocPortable);
     }
 
-    const unsigned char *src_ptr = reinterpret_cast<const unsigned char *>(src);
+    const auto *src_ptr = reinterpret_cast<const unsigned char *>(src);
```
Comment (stsoe) [inline src/runtime_src/hip/core/memory.cpp:106]:
```suggestion
    auto src_ptr = reinterpret_cast<const unsigned char *>(src);
```
Already const and ptr per cast.

File: src/runtime_src/hip/core/memory.cpp [hunk 2]
```diff
@@ -119,21 +119,21 @@ namespace xrt::core::hip
         // pinned hip mem
         assert(dst_hip_mem->get_flags() == hipHostMallocDefault || dst_hip_mem->get_flags() == hipHostMallocPortable);
     }
-    unsigned char *dst_ptr = reinterpret_cast<unsigned char *>(dst);
+    auto *dst_ptr = reinterpret_cast<unsigned char *>(dst);
```
Comment (stsoe) [inline src/runtime_src/hip/core/memory.cpp:122]:
```suggestion
    auto dst_ptr = reinterpret_cast<unsigned char *>(dst);
```

**Review by stsoe — CHANGES_REQUESTED** [review]

File: src/runtime_src/hip/core/memory.cpp (hunk 1 above)
Comment (sonals) [reply src/runtime_src/hip/core/memory.cpp]: Updated

File: src/runtime_src/hip/core/memory.cpp (hunk 2 above)
Comment (sonals) [reply src/runtime_src/hip/core/memory.cpp]: Updated

**Pull request merged**